Merge Review composer UX - Meeds-io/MIPs#54

MayTekayaa opened the pull request:
This change will merge the review composer UX MIP.

MayTekayaa committed:
fix: Remove placeholder when editing an activity - MEED-2444 - Meeds-io/MIPs#54

This change will adjust the box of the send kudos action inside of the composer extension.

MayTekayaa committed:
fix: Fix the send kudos button activation - MEED-2443 - Meeds-io/MIPs#54 (#335)

This change will fix the send kudos button activation.

MayTekayaa committed:
fix: Improve the toolbar post CTA - MEED-2401 - Meeds-io/MIPs#54 (#336)

This change will adjust the composer send kudos button.

MayTekayaa committed:
feat: Add the ability to send a kudos from the composer - MEED-2071 - Meeds-io/MIPs#54 (#338)

This change will add the ability to send a kudos from the composer .

MayTekayaa committed:
feat: Add CTAs below the post CTA - MEED-2072 - Meeds-io/MIPs#54 (#339)

This fix will change the display of the send kudos action used in the composer.

MayTekayaa committed:
feat: Disable send kudos button when no receiver has been added - MEED-2418 - Meeds-io/MIPs#54 (#340)

This change will disable the send kudos button when no receiver has been added.

MayTekayaa committed:
fix: Update the receiver field after sending kudos - MEED-2419 - Meeds-io/MIPs#54 (#345)

This change will update the receiver field after sending kudos.

MayTekayaa committed:
feat: Adjust the CTAs display from the general stream - MEED-2073 - Meeds-io/MIPs#54 (#352)

This change will display the composer send kudos extensions when only the user is in the space stream.

MayTekayaa committed:
feat: Add the ability to send a kudos from genral stream- MEED-2076 - Meeds-io/MIP#54

This change will add the ability to send a kudos from genral stream.

MayTekayaa committed:
fix: Adjust the kudos CTA from general stream - MEED-2568 - Meeds-io/MIP#54 (#367)

This change will adjust the toolbar send kudos component.

MayTekayaa committed:
fix: Fix the padding between the audience and chosen audience - MEED-2580 - Meeds-io/MIP#54 (#369)

This change will fix the padding between the audience and chosen audience.

MayTekayaa committed:
fix: Add some enhancements to the the activity toolbar icons - MEED-2581 - Meeds-io/MIP#54

This change will add some enhancements to the the toolbar activity icons.

MayTekayaa committed:
fix: Clean the kudos drawer after sending one to a user - MEED-2585 - Meeds-io/MIP#54 (#372)

This change will clean the kudos drawer after sending one to a user.

MayTekayaa committed:
feat: Add a tooltip in the kudos drawer for reset audience icon - MEED-2584 - Meeds-io/MIP#54 (#373)

This change will add a tooltip in the kudos drawer for reset audience icon.

MayTekayaa committed:
fix: Review the display of the composer drawer when space audience chosen - MEED-2569 - Meeds-io/MIP#54 (#374)

This change will add some adjustments to the kudos drawer audience.

rdenarie commented on kudos-webapps/src/main/webapp/vue-app/js/Kudos.js:
```diff
@@ -219,6 +219,19 @@ export function registerOverviewExtension() {
 }
 
+export function registerComposerExtension() {
+  extensionRegistry.registerComponent('ComposerAction', 'composer-action-item', {
```
Why kudos is a composer-action-item as poll is a activity-composer-footer-action ? I think it should be the same

MayTekayaa replied:
"activity-composer-footer-action" is an old extension created when the poll module was added,
and "composer-action-item" is a new extension created to be used in other components.
I think we should keep both extensions for now.

rdenarie replied:
In that case, why extension for kudos is in composer-action-item and poll is in activity-composer-footer-action ?
Should we put both in same extension ? 
The idea is to have a homogeneous behaviour

MayTekayaa replied:
Yeah, I know, I had created the registry extension for kudos to be able to use it in other components.

rdenarie commented on kudos-webapps/src/main/webapp/vue-app/kudos/components/SendKudosComposer.vue:
```diff
@@ -0,0 +1,38 @@
+<template>
```
Should we add a header ?

MayTekayaa replied:
yes!

rdenarie commented on kudos-webapps/src/main/webapp/vue-app/kudos/components/SendKudosToolbarAction.vue:
```diff
@@ -0,0 +1,59 @@
+<template>
```
Should we add a header ?

MayTekayaa replied:
yes!

MayTekayaa committed:
fix: Adjust the display of icons used in the composer drawer - MEED-2606 -Meeds-io/MIP#54 (#378)

This change will adjust the alignment of the action title in composer drawer.

MayTekayaa committed:
fix: Align action labels below the composer - MEED-2609 -Meeds-io/MIP#54 (#381)

This change will align action labels below the composer and add the license header for new components.
